content/blog/2021/09/towards-a-universal-policy-platform.md

```diff
@@ -0,0 +1,155 @@
+to perform String, Encoding, Regex, ... operations.
+When building a Rego policy into a WebAssembly module, some of these built-in
+functions are going to be implemented inside of the Wasm file itself; while others
+have to be provided at execution time by the WebAssembly runtime evaluating the module.
+
+The Burrego library implements the built-in functions that are not supported natively
+by the WebAssembly modules produced by `opa build`. There are still some built-ins that
+are not yet provided by Burrego, however, based on the policies we have seen in the open,
+the ones we already support should be enough for the majority of Kubernetes users.
+
+[This GitHub issue](https://github.com/kubewarden/policy-evaluator/issues/56)
+keeps track of the Rego built-ins we have still to implement. Feel free to
+comment over there to prioritize our work.
+
+## One language, two frameworks
+
+Open Policy Agent integrates with Kubernetes using the [kube-mgmt](https://github.com/open-policy-agent/kube-mgmt)
+component.
+More recently, a new Kubernetes integration gained popularity: [Gatekeeper](https://github.com/open-policy-agent/gatekeeper).
+
+Both kube-mgmt and Gatekeeper use Rego to write their policies. However
+the way input parameters and other important information is exposed to the policy,
+and the way the policy has to give its answer differs between the two.
+
+This leads to the unfortunate situation where a Kubernetes policy written for
+the OPA runtime will not work as expected if enforced by Gatekeeper, and vice versa.
+
+We don't like this fragmentation in the "Rego landscape", hence we worked to ensure
+that Kubewarden can execute both policies built for OPA and for Gatekeeper
+-- **without any change needed on existing policies**.
+
+## Authoring a Rego policy
+
+Rego polices can be built into a single WebAssembly module by using the `opa build`
+command. No change is required to already existing Rego policies.
+
+We want these WebAssembly modules to be loaded by Kubewarden in a transparent way,
+the end users of your policy will not have to deal with the subtle differences
+between a "Kubewarden native" policy, an OPA or a Gatekeeper one.
+To achieve this portability, the WebAssembly module must be enriched with some
+Kubewarden-specific metadata.
+
+Adding metadata to a WebAssembly module produced by `opa build` works in the
+same ways as with modules produced by Rust, Go, ... This is done using the
+`kwctl annotate` command.
+
+Finally, the annotated file can be pushed to a OCI registry with the usual
+`kwctl push` command.
+
+Policy authors can also evaluate their policy locally, before pushing it, via
+the `kwctl run` command.
+
+If you are already familiar with Kubewarden, you will notice the developer
+workflow stays the same, regardless of the programming language used.
+
+All these steps are described in detail inside of our [documentation](TODO).
```

Comment:
Replace TODO

Reply:
Yes, but only after https://github.com/kubewarden/docs/pull/48 is merged and published

Reply:
We can populate that here and only merge after the docs PR is merged, the URI would be https://docs.kubewarden.io/writing-policies/rego/01-intro.html

Reply:
thanks, I've added it